### PR TITLE
[DOCS] Updating references to docs 2024

### PR DESCRIPTION
**Pull request by sgolebiewski-intel**
Updating references to `docs.openvino.ai` 2024 version.

### CODE REVIEW

File: notebooks/104-model-tools/104-model-tools.ipynb [hunk 1]
```diff
@@ -331,11 +331,11 @@
...
       "[ INFO ] The model was converted to IR v11, the latest model format that corresponds to the source DL framework input/output format. While IR v11 is backwards compatible with OpenVINO Inference Engine API v1.0, please use API v2.0 (as of 2022.1) to take advantage of the latest improvements in IR v11.\n",
-      "Find more information about API v2.0 and IR v11 at https://docs.openvino.ai/2023.0/openvino_2_0_transition_guide.html\n",
+      "Find more information about API v2.0 and IR v11 at https://docs.openvino.ai/2023.3/openvino_2_0_transition_guide.html\n",
       "[ INFO ] MO command line tool is considered as the legacy conversion API as of OpenVINO 2023.2 release. Please use OpenVINO Model Converter (OVC). OVC represents a lightweight alternative of MO and provides simplified model conversion API. \n",
-      "Find more information about transition from MO to OVC at https://docs.openvino.ai/2023.2/openvino_docs_OV_Converter_UG_prepare_model_convert_model_MO_OVC_transition.html\n",
+      "Find more information about transition from MO to OVC at https://docs.openvino.ai/2023.3/openvino_docs_OV_Converter_UG_prepare_model_convert_model_MO_OVC_transition.html\n",
```
Comment (msmykx-intel) [inline notebooks/104-model-tools/104-model-tools.ipynb:338]: We have this article in 2024 -> https://docs.openvino.ai/2024/documentation/legacy-features/transition-legacy-conversion-api.html

I think we should link it to 2024, unless there is a reason we are linking this to 23.3?

File: notebooks/108-gpu-device/108-gpu-device.ipynb [hunk 2]
```diff
@@ -818,7 +818,7 @@
...
+    "The latency and throughput hints mentioned above are great and can make a difference when used adequately but they usually use just one device, either due to the [AUTO plugin](https://docs.openvino.ai/2024/openvino-workflow/running-inference/inference-devices-and-modes/auto-device-selection.html#how-auto-works) or by manual specification of the device name as above. When we have multiple devices, such as an integrated and discrete GPU, we may use both at the same time to improve the utilization of the resources. In order to do this, OpenVINO provides a virtual device called [MULTI](https://docs.openvino.ai/nightly/openvino_docs_OV_UG_Running_on_multiple_devices.html), which is just a combination of the existent devices that knows how to split inference work between them, leveraging the capabilities of each device.\n",
```
Comment (msmykx-intel) [inline notebooks/108-gpu-device/108-gpu-device.ipynb:821]:
```suggestion
    "The latency and throughput hints mentioned above are great and can make a difference when used adequately but they usually use just one device, either due to the [AUTO plugin](https://docs.openvino.ai/2024/openvino-workflow/running-inference/inference-devices-and-modes/auto-device-selection.html#how-auto-works) or by manual specification of the device name as above. When we have multiple devices, such as an integrated and discrete GPU, we may use both at the same time to improve the utilization of the resources. In order to do this, OpenVINO provides a virtual device called [MULTI](https://docs.openvino.ai/2024/openvino-workflow/running-inference/inference-devices-and-modes/multi-device.html), which is just a combination of the existent devices that knows how to split inference work between them, leveraging the capabilities of each device.\n",
```

File: notebooks/104-model-tools/104-model-tools.ipynb (hunk 1 above)
Comment (sgolebiewski-intel) [reply notebooks/104-model-tools/104-model-tools.ipynb:338]: Yeah, let's update with a link to 2024

File: notebooks/113-image-classification-quantization/113-image-classification-quantization.ipynb [hunk 3]
```diff
@@ -236,7 +236,7 @@
     "## Run nncf.quantize for Getting an Optimized Model\n",
     "[back to top ⬆️](#Table-of-contents:)\n",
     "\n",
-    "`nncf.quantize` function accepts model and prepared quantization dataset for performing basic quantization. Optionally, additional parameters like `subset_size`, `preset`, `ignored_scope` can be provided to improve quantization result if applicable. More details about supported parameters can be found on this [page](https://docs.openvino.ai/2023.3/basic_quantization_flow.html#tune-quantization-parameters)"
+    "`nncf.quantize` function accepts model and prepared quantization dataset for performing basic quantization. Optionally, additional parameters like `subset_size`, `preset`, `ignored_scope` can be provided to improve quantization result if applicable. More details about supported parameters can be found on this [page](https://docs.openvino.ai/2024/openvino-workflow/model-optimization-guide/quantizing-models-post-training/basic-quantization-flow.html)"
```
Comment (msmykx-intel) [inline notebooks/113-image-classification-quantization/113-image-classification-quantization.ipynb:239]:
```suggestion
    "`nncf.quantize` function accepts model and prepared quantization dataset for performing basic quantization. Optionally, additional parameters like `subset_size`, `preset`, `ignored_scope` can be provided to improve quantization result if applicable. More details about supported parameters can be found on this [page](https://docs.openvino.ai/2024/openvino-workflow/model-optimization-guide/quantizing-models-post-training/basic-quantization-flow.html#tune-quantization-parameters)"
```

File: notebooks/107-speech-recognition-quantization/107-speech-recognition-quantization-data2vec.ipynb [hunk 4]
```diff
@@ -497,7 +497,7 @@
...
+    "2. Run `nncf.quantize` for getting an optimized model. The `nncf.quantize` function provides an interface for model quantization. It requires an instance of the OpenVINO Model and quantization dataset. Optionally, some additional parameters for the configuration quantization process (number of samples for quantization, preset, ignored scope, etc.) can be provided. For more accurate results, we should keep the operation in the postprocessing subgraph in floating point precision, using the `ignored_scope` parameter. For more information see [Tune quantization parameters](https://docs.openvino.ai/2024/openvino-workflow/model-optimization-guide/quantizing-models-post-training/basic-quantization-flow.html).\n",
```
Comment (msmykx-intel) [inline notebooks/107-speech-recognition-quantization/107-speech-recognition-quantization-data2vec.ipynb:500]:
```suggestion
    "2. Run `nncf.quantize` for getting an optimized model. The `nncf.quantize` function provides an interface for model quantization. It requires an instance of the OpenVINO Model and quantization dataset. Optionally, some additional parameters for the configuration quantization process (number of samples for quantization, preset, ignored scope, etc.) can be provided. For more accurate results, we should keep the operation in the postprocessing subgraph in floating point precision, using the `ignored_scope` parameter. For more information see [Tune quantization parameters](https://docs.openvino.ai/2024/openvino-workflow/model-optimization-guide/quantizing-models-post-training/basic-quantization-flow.html#tune-quantization-parameters).\n",
```

File: notebooks/107-speech-recognition-quantization/107-speech-recognition-quantization-wav2vec2.ipynb [hunk 5]
```diff
@@ -214,7 +214,7 @@
...
+    "2. Run `nncf.quantize` for getting an optimized model. The `nncf.quantize` function provides an interface for model quantization. It requires an instance of the OpenVINO Model and quantization dataset. Optionally, some additional parameters for the configuration quantization process (number of samples for quantization, preset, ignored scope, etc.) can be provided. For more accurate results, we should keep the operation in the postprocessing subgraph in floating point precision, using the `ignored_scope` parameter. For more information see [Tune quantization parameters](https://docs.openvino.ai/2024/openvino-workflow/model-optimization-guide/quantizing-models-post-training/basic-quantization-flow.html). For this model, ignored scope was selected experimentally, based on result of quantization with accuracy control. For understanding how it works please check following [notebook](../122-quantizing-model-with-accuracy-control/122-speech-recognition-quantization-wav2vec2.ipynb)\n",
```
Comment (msmykx-intel) [inline notebooks/107-speech-recognition-quantization/107-speech-recognition-quantization-wav2vec2.ipynb:217]:
```suggestion
    "2. Run `nncf.quantize` for getting an optimized model. The `nncf.quantize` function provides an interface for model quantization. It requires an instance of the OpenVINO Model and quantization dataset. Optionally, some additional parameters for the configuration quantization process (number of samples for quantization, preset, ignored scope, etc.) can be provided. For more accurate results, we should keep the operation in the postprocessing subgraph in floating point precision, using the `ignored_scope` parameter. For more information see [Tune quantization parameters](https://docs.openvino.ai/2024/openvino-workflow/model-optimization-guide/quantizing-models-post-training/basic-quantization-flow.html#tune-quantization-parameters). For this model, ignored scope was selected experimentally, based on result of quantization with accuracy control. For understanding how it works please check following [notebook](../122-quantizing-model-with-accuracy-control/122-speech-recognition-quantization-wav2vec2.ipynb)\n",
```

File: notebooks/302-pytorch-quantization-aware-training/302-pytorch-quantization-aware-training.ipynb [hunk 6]
```diff
@@ -725,7 +725,7 @@
    "source": [
     "2. Run `nncf.quantize` for Getting an Optimized Model.\n",
     "\n",
-    "`nncf.quantize` function accepts model and prepared quantization dataset for performing basic quantization. Optionally, additional parameters like `subset_size`, `preset`, `ignored_scope` can be provided to improve quantization result if applicable. More details about supported parameters can be found on this [page](https://docs.openvino.ai/2023.3/basic_quantization_flow.html#tune-quantization-parameters)"
+    "`nncf.quantize` function accepts model and prepared quantization dataset for performing basic quantization. Optionally, additional parameters like `subset_size`, `preset`, `ignored_scope` can be provided to improve quantization result if applicable. More details about supported parameters can be found on this [page](https://docs.openvino.ai/2024/openvino-workflow/model-optimization-guide/quantizing-models-post-training/basic-quantization-flow.html)"
```
Comment (msmykx-intel) [inline notebooks/302-pytorch-quantization-aware-training/302-pytorch-quantization-aware-training.ipynb:728]:
```suggestion
    "`nncf.quantize` function accepts model and prepared quantization dataset for performing basic quantization. Optionally, additional parameters like `subset_size`, `preset`, `ignored_scope` can be provided to improve quantization result if applicable. More details about supported parameters can be found on this [page](https://docs.openvino.ai/2024/openvino-workflow/model-optimization-guide/quantizing-models-post-training/basic-quantization-flow.html#tune-quantization-parameters)"
```

File: notebooks/124-hugging-face-hub/124-hugging-face-hub.ipynb [hunk 7]
```diff
@@ -179,7 +179,7 @@
    "source": [
     "### Converting the Model to OpenVINO IR format\n",
     "[back to top ⬆️](#Table-of-contents:)\n",
-    "We use the OpenVINO [Model conversion API](https://docs.openvino.ai/2023.3/openvino_docs_model_processing_introduction.html#convert-a-model-in-python-convert-model) to convert the model (this one is implemented in PyTorch) to OpenVINO Intermediate Representation (IR).\n",
+    "We use the OpenVINO [Model conversion API](https://docs.openvino.ai/2024/openvino-workflow/model-preparation.html#convert-a-model-in-python-convert-model) to convert the model (this one is implemented in PyTorch) to OpenVINO Intermediate Representation (IR).\n",
```
Comment (msmykx-intel) [inline notebooks/124-hugging-face-hub/124-hugging-face-hub.ipynb:182]:
```suggestion
    "We use the OpenVINO [Model conversion API](https://docs.openvino.ai/2024/openvino-workflow/model-preparation.html#convert-a-model-with-python-convert-model) to convert the model (this one is implemented in PyTorch) to OpenVINO Intermediate Representation (IR).\n",
```

File: notebooks/257-llava-multimodal-chatbot/257-llava-multimodal-chatbot.ipynb [hunk 8]
```diff
@@ -219,7 +219,7 @@
...
+    "The code below prepares function for converting LLaVA model to OpenVINO Intermediate Representation format. It splits model on parts described above, prepare example inputs for each part and convert each part using [OpenVINO Model Conversion API](https://docs.openvino.ai/2024/openvino-workflow/model-preparation.html#convert-a-model-in-python-convert-model). `ov.convert_model` function accepts PyTorch model instance and returns `ov.Model` object that represent model in OpenVINO format. It is ready to use for loading on device using `ov.compile_model` or can be saved on disk using `ov.save_model`."
```
Comment (msmykx-intel) [inline notebooks/257-llava-multimodal-chatbot/257-llava-multimodal-chatbot.ipynb:222]:
```suggestion
    "The code below prepares function for converting LLaVA model to OpenVINO Intermediate Representation format. It splits model on parts described above, prepare example inputs for each part and convert each part using [OpenVINO Model Conversion API](https://docs.openvino.ai/2024/openvino-workflow/model-preparation.html#convert-a-model-with-python-convert-model). `ov.convert_model` function accepts PyTorch model instance and returns `ov.Model` object that represent model in OpenVINO format. It is ready to use for loading on device using `ov.compile_model` or can be saved on disk using `ov.save_model`."
```

File: notebooks/257-llava-multimodal-chatbot/257-videollava-multimodal-chatbot.ipynb [hunk 9]
```diff
@@ -244,7 +244,7 @@
...
+    "The code below prepares function for converting Video-LLaVA model to OpenVINO Intermediate Representation format. It splits model on parts described above, prepare example inputs for each part and convert each part using [OpenVINO Model Conversion API](https://docs.openvino.ai/2024/openvino-workflow/model-preparation.html#convert-a-model-in-python-convert-model). `ov.convert_model` function accepts PyTorch model instance and returns `ov.Model` object that represent model in OpenVINO format. It is ready to use for loading on device using `ov.compile_model` or can be saved on disk using `ov.save_model`."
```
Comment (msmykx-intel) [inline notebooks/257-llava-multimodal-chatbot/257-videollava-multimodal-chatbot.ipynb:247]:
```suggestion
    "The code below prepares function for converting Video-LLaVA model to OpenVINO Intermediate Representation format. It splits model on parts described above, prepare example inputs for each part and convert each part using [OpenVINO Model Conversion API](https://docs.openvino.ai/2024/openvino-workflow/model-preparation.html#convert-a-model-with-python-convert-model). `ov.convert_model` function accepts PyTorch model instance and returns `ov.Model` object that represent model in OpenVINO format. It is ready to use for loading on device using `ov.compile_model` or can be saved on disk using `ov.save_model`."
```

File: notebooks/276-stable-diffusion-torchdynamo-backend/276-stable-diffusion-torchdynamo-backend.ipynb [hunk 10]
```diff
@@ -145,14 +145,14 @@
...
     "   - compiled by TorchDynamo and “flattened”,\n",
     "   - falling back to the eager-mode, due to unsupported Python constructs (like control-flow code).\n",
     "2. Graph lowering - all PyTorch operations are decomposed into their constituent kernels specific to the chosen backend.\n",
     "3. Graph compilation - the kernels call their corresponding low-level device-specific operations.\n",
     "\n",
-    "Select device for inference and enable or disable saving the optimized model files to a hard drive, after the first application run. This makes them available for the following application executions, reducing the first-inference latency. Read more about available [Environment Variables options](https://docs.openvino.ai/2023.3/pytorch_2_0_torch_compile.html#environment-variables)"
+    "Select device for inference and enable or disable saving the optimized model files to a hard drive, after the first application run. This makes them available for the following application executions, reducing the first-inference latency. Read more about available [Environment Variables options](https://docs.openvino.ai/2024/openvino-workflow/torch-compile.html#environment-variables)"
```
Comment (msmykx-intel) [inline notebooks/276-stable-diffusion-torchdynamo-backend/276-stable-diffusion-torchdynamo-backend.ipynb:155]:
```suggestion
    "Select device for inference and enable or disable saving the optimized model files to a hard drive, after the first application run. This makes them available for the following application executions, reducing the first-inference latency. Read more about available [Environment Variables options](https://docs.openvino.ai/2024/openvino-workflow/torch-compile.html#options)"
```

Environment Variables were changed from it's own part of an article into a note, linking now "Options" which describe them in current release

**Review by eaidova — APPROVED** [review]

**Comment by eaidova** [comment]
@sgolebiewski-intel  @msmykx-intel  is it ready to merge?

**Comment by sgolebiewski-intel** [comment]
> @sgolebiewski-intel @msmykx-intel is it ready to merge?

I have rebased it, just in case and it's ready to merge :)

**Pull request merged**